### ISSUE TITLE
[BUG] FunctionHook Fallback Behavior

### ISSUE THREAD

**Issue by Daztek** [nwnxee/unified#397]
Due to #381 events now request their hooks when they're subscribed to, usually during the module load script. 

Which means the following fallback behavior in for example [NWNX_Rename](https://github.com/nwnxee/unified/blob/master/Plugins/Rename/Rename.cpp#L77-L90) does not work as intended anymore:

```c++
try //try to request a hook. If it fails that means that NWNX_EVENTS_PARTY_EVENTS is enabled, which should broadcast a message to trigger the name change anyway.
{
	GetServices()->m_hooks->RequestSharedHook<Functions::CNWSMessage__HandlePlayerToServerParty,int32_t,CNWSMessage*,CNWSPlayer*,unsigned char>(&HookPartyInvite);
}
catch (...){}
GetServices()->m_messaging->SubscribeMessage("NWNX_EVENT_SIGNAL_EVENT_SKIPPED",
	[this](const std::vector<std::string> message)
	{
		if (message[1] == "0") //if the event was skipped it doesn't matter what the event was.
		{
			if (message[0] == "NWNX_ON_PARTY_INVITE_BEFORE") this->GlobalNameChange(false, nullptr);
			if (message[0] == "NWNX_ON_PARTY_INVITE_AFTER") this->GlobalNameChange(true, nullptr);
		}
	});
```

NWNX_Rename will now apply its hook first which causes the NWNX_Events hook to fail.

**Comment by plenarius** [comment]
Any suggestions on correcting this while I'm working in `NWNX_Rename`?

**Comment by mtijanic** [comment]
Merged into #627
